source/fundamentals/data-formats/serialization.txt

```diff
@@ -0,0 +1,178 @@
+with configurations to encode defaults, encode nulls, and define class 
+discriminators.
+
+.. note::
+
+   To use the ``Codec`` interface to specify custom encoding and decoding of 
+   Kotlin objects to BSON data, see the :ref:`Codecs <fundamentals-codecs>` 
+   documentation.
+
+Although you can use the Kotlin driver with the Kotlin serialization ``Json`` 
+library, the ``Json`` serializer does *not* directly support BSON value types such
+as ``ObjectId``. You must provide a custom serializer that can handle the 
+conversion between BSON and JSON. 
+
+Supported Types
+~~~~~~~~~~~~~~~
+
+The Kotlin driver supports: 
+
+- All Kotlin types that are supported by the Kotlin serialization library 
+- All available :manual:`BSON types </reference/bson-types>`
+
+You can implement a custom serializers to handle any other types. See the 
+`official Kotlin Documentation <https://kotlinlang.org/docs/serialization.html>`__ 
+for more information.
+
+Add Kotlin Serialization to Your Project
+----------------------------------------
+
+The Kotlin driver serialization support depends on the the official Kotlin 
+serialization library. You must add `Kotlin Serialization <https://github.com/Kotlin/kotlinx.serialization>`__ 
+to your project:
+
+.. tabs::
+
+   .. tab::
+      :tabid: Gradle
+
+      If you are using `Gradle <https://gradle.org/>`__ to manage your 
+      dependencies, add the following to your ``build.gradle.kts`` dependencies list:
+
+      .. code-block:: kotlin
+        :caption: build.gradle.kts
+
+        implementation("org.jetbrains.kotlinx:kotlinx-serialization-core:{+serializationVersion+}")
+
+   .. tab::
+      :tabid: Maven
+
+      If you are using `Maven <https://maven.apache.org/>`__ to manage your 
+      dependencies, add the following to your ``pom.xml`` dependencies list:
+
+      .. code-block:: kotlin
+        :caption: pom.xml
+   
+        <dependency>
+            <groupId>org.jetbrains.kotlinx</groupId>
+            <artifactId>kotlinx-serialization-core</artifactId>
+            <version>{+serializationVersion+}</version>
+        </dependency>
+
+Annotate Data Classes
+---------------------
+
+To declare a class as serializable, annotate your Kotlin data classes with the 
+``@Serializable`` annotation that comes from the Kotlin serialization framework.
+
+You can use your data classes in your code as normal after you mark them as serializable.
+The Kotlin driver and the Kotlin Serialization framework will handle the 
+BSON serialization and deserialization.
+
+This example shows a simple data class annotated with the following: 
+
+- ``@Serializable`` to mark the class as serializable
+- ``@SerialName`` to specify the name of the ``id`` and ``manufacturer`` properties 
+   in the BSON document. This can be used in place of the ``@BsonId`` and 
+   ``@BsonProperty`` annotations, which are unsupported in serializable classes.
+- ``@Contextual`` to mark the BSON ``id`` property to use the ``ObjectIdSerializer``.
```

Comment:
Suggest: property to use the builtin ``ObjectIdSerializer``.

Reply:
Updated